src/main/java/org/rsa/net/apis/discourse/DiscourseAPI.java

```diff
@@ -0,0 +1,69 @@
+package org.rsa.net.apis.discourse;
+
+import com.google.common.util.concurrent.RateLimiter;
+import lombok.Getter;
+import org.rsa.net.apis.discourse.domain.Category;
+import org.rsa.net.apis.discourse.domain.Topic;
+import org.rsa.net.apis.discourse.models.CategoryModel;
+import org.rsa.net.apis.discourse.models.CategoryTopicsModel;
+import org.rsa.net.apis.discourse.models.SiteBasicInfoModel;
+import org.rsa.net.apis.discourse.models.TopicModel;
+import org.rsa.net.HttpClient;
+import org.rsa.net.apis.discourse.transformers.CategoryTransformer;
+import org.rsa.net.apis.discourse.transformers.TopicTransformer;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+public class DiscourseAPI {
+    private static final RateLimiter rateLimiter = RateLimiter.create(0.4); // 1 API hit every 3 seconds
+    private static final @Getter String BASE_URL = "https://devforum.roblox.com";
+    private static final String CATEGORIES_URL = BASE_URL + "/site.json";
+    private static final String CATEGORY_URL_FORMAT = BASE_URL + "/c/%s.json";
+    private static final String TOPICS_URL_FORMAT = BASE_URL + "/t/%s.json";
+    private static final String SITE_BASIC_INFO_URL = BASE_URL + "/site/basic-info.json";
+    private static Map<String, Category> latestCategoryInformation;
+
+    public static SiteBasicInfoModel fetchSiteBasicInfo() throws IOException {
+        rateLimiter.acquire();
+        return HttpClient.get(SITE_BASIC_INFO_URL, SiteBasicInfoModel.class);
+    }
+
+    public static Map<String, Category> fetchAllCategoryInformation() throws IOException {
+        rateLimiter.acquire();
+        CategoryModel responseObj = HttpClient.get(CATEGORIES_URL, CategoryModel.class);
+        Map<String, Category> categoryDetails = CategoryTransformer.fromResponse(responseObj);
+        return categoryDetails;
+    }
+
+    public static Optional<String> fetchLatestNonPinnedTopicId(String categoryId) throws IOException {
```

Comment:
It would be easier to mock and test these if they weren't static. Same for HttpClient

Reply:
Tests aren't really a primary focus here. I can't say I've seen a Discord bot with tests. 😅

Reply:
However, if / when we get to DI, the statics will make that a fair bit harder.